lib/getHashDigest.js

```diff
@@ -120,12 +120,17 @@ function getHashDigest(buffer, algorithm, digestType, maxLength) {
     digestType === "base49" ||
     digestType === "base52" ||
     digestType === "base58" ||
-    digestType === "base62"
+    digestType === "base62" ||
+    digestType === "base64safe"
   ) {
-    return encodeBufferToBase(hash.digest(), digestType.substr(4), maxLength);
-  } else {
-    return hash.digest(digestType || "hex").substr(0, maxLength);
+    return encodeBufferToBase(
+      hash.digest(),
+      digestType === "base64safe" ? 64 : digestType.substr(4),
```

Comment:
Let's also use a table to achieve higher entropy, removing characters can create conflicts, i.e. `3ZWmHo0=hPMWE2rZeN_oHB6` and `3ZWmHo0hPMWE2rZe=N_oHB6` will be same after removing characters, this is bad